components/hubspot/sources/new-contact-in-list/new-contact-in-list.js

```diff
@@ -1,50 +1,60 @@
-const hubspot = require("../../hubspot.app.js");
+const common = require("../common.js");
 
 module.exports = {
+  ...common,
   key: "hubspot-new-contact-in-list",
   name: "New Contact in List",
   description: "Emits an event for each new contact in a list.",
-  version: "0.0.1",
+  version: "0.0.2",
   dedupe: "unique",
   props: {
-    hubspot,
-    lists: {
-      type: "string[]",
-      label: "Lists",
-      optional: false,
-      async options() {
-        const results = await this.hubspot.getLists();
-        const options = results.map((result) => {
-          const label = result.name;
-          return {
-            label,
-            value: JSON.stringify({ label, value: result.listId }),
-          };
-        });
-        return options;
-      },
-    },
-    db: "$.service.db",
-    timer: {
-      type: "$.interface.timer",
-      default: {
-        intervalSeconds: 60 * 15,
-      },
+    ...common.props,
+    lists: { propDefinition: [common.props.hubspot, "lists"] },
+  },
+  hooks: {
+    async deploy() {
+      // By default, only a limited set of properties are returned from the API.
+      // Get all possible contact properties to request for each contact.
+      const properties = await this.hubspot.createPropertiesArray();
+      this.db.set("properties", properties);
     },
```

Comment:
Maybe we can put this hook in the `common` module?

Reply:
Updated